subsys/bluetooth/services/fast_pair/fp_auth.c

```diff
@@ -80,12 +81,21 @@ enum bt_security_err auth_pairing_accept(struct bt_conn *conn,
 					 const struct bt_conn_pairing_feat *const feat)
 {
 	bool conn_handled = is_conn_handled(conn);
+	bool io_capability_valid = (feat->io_capability != BT_IO_NO_INPUT_OUTPUT);
 
 	if (!conn_handled) {
 		LOG_ERR("Invalid conn (pairing accept): %p", (void *)conn);
 	}
 
-	return conn_handled ? BT_SECURITY_ERR_SUCCESS : BT_SECURITY_ERR_UNSPECIFIED;
+	if (!io_capability_valid) {
+		LOG_ERR("Invalid IO capability (pairing accept): %p", (void *)conn);
+	}
+
+	if (conn_handled && io_capability_valid) {
+		return BT_SECURITY_ERR_SUCCESS;
+	} else {
+		return BT_SECURITY_ERR_UNSPECIFIED;
```

Comment:
I think it would be better to return `BT_SECURITY_ERR_AUTH_REQUIREMENT` which should be returned when for example IO capabilities cannot be met if I understand your case correctly

Reply:
Ok, changed